client/components/social-buttons/facebook.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import SocialLogo from 'social-logos';
+import { loadScript } from 'lib/load-script';
+
+export default class FacebookLoginButton extends Component {
+
+	// See: https://developers.facebook.com/docs/javascript/reference/FB.init/v2.8
+	static propTypes = {
+		appId: PropTypes.string.isRequired,
+		version: PropTypes.string,
+		cookie: PropTypes.bool,
+		status: PropTypes.bool,
+		xfbml: PropTypes.bool,
+		responseHandler: PropTypes.func.isRequired,
+		scope: PropTypes.string,
+	};
+
+	static defaultProps = {
+		version: 'v2.8',
+		cookie: false,
+		status: false,
+		xfbml: true,
+		scope: 'public_profile,email',
+	};
+
+	constructor( props ) {
+		super( props );
+		this.FB = null;
+		this.handleClick = this.handleClick.bind( this );
+	}
+
+	componentWillMount() {
+		this.loadFacebookAuth();
+	}
+
+	loadDependency() {
+		if ( window.FB ) {
+			return Promise.resolve( window.FB );
+		}
+
+		return new Promise( resolve => {
+			loadScript( '//connect.facebook.net/en_US/sdk.js', () => resolve( window.FB ) );
+		} );
+	}
+
+	loadFacebookAuth() {
+		if ( this.FB ) {
+			return Promise.resolve();
+		}
+
+		return new Promise( resolve => {
+			this.loadDependency().then( FB => {
+				FB.init( {
+					appId: this.props.appId,
+					version: this.props.version,
+					cookie: this.props.cookie,
+					xfbml: this.props.xfbml,
+				} );
+				this.FB = FB;
+				resolve();
+			} );
+		} );
+	}
+
+	handleClick() {
+		const { responseHandler, scope } = this.props;
+
+		// Handle click async if the library is not loaded yet
+		// the popup might be blocked by the browser in that case
+		this.loadFacebookAuth().then( () => {
```

Comment:
Instead of creating a new promise in this case, could we return the existing promise that was created in `componentWillMount`? Otherwise, we'll be calling `loadScript` twice and (I believe) injecting an extra `<script>` into the header.